### ISSUE TITLE
`parse_generics_shim!` fails to parse complex generic constraints

### ISSUE THREAD

**Issue by Limeth** [DanielKeep/rust-parse-generics#1]
I have been working on adding generic constraints to [the mopa crate](https://github.com/chris-morgan/mopa) and I've encountered a problem with your `parse_generics_shim!` macro.

This would work:

``` rust
parse_generics_shim! {
        { .. },
        then $callback_name ! ( $($callback_args)* ),
        <X, F: Float, D: Deep<F> >
}
```

This would **not** work:

``` rust
parse_generics_shim! {
        { .. },
        then $callback_name ! ( $($callback_args)* ),
        <X, F: Float, D: Deep<F>>
}
```

Notice the lack of a space at the end of generics. It seems to take the `>>` as a single tag, the bitshift operator.

@bjorn3 has done some great work on solving this issue, see [this commit](https://github.com/Limeth/mopa/commit/e118ccb0ada47719cb0d4386c9a8fd00494d123a). I think this code should be in your macro instead of on the side of the user.

**Comment by bjorn3** [comment]
I have published a crate for splitting the >> token on crates.io: [crates.io/crates/split_tokens](crates.io/crates/split_tokens)

**Comment by DanielKeep** [comment]
That would consume recursion levels equal to the number of input tokens.  That's _really_ untenable given how complicated these macros are already.

In fact, the parser was _already_ dealing with `>>` in constraints.  It just didn't have a rule to deal with `>>` terminating a constraint _and_ a generic parameter list in the same token.

Published v0.1.1 with this fix.
